packages/jest-phabricator-coverage/src/index.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) 2014, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ * @flow
+ */
+
+'use strict';
+
+import type {
+  AggregatedResult,
+  CoverageMap,
+  FormattedTestResult,
+} from 'types/TestResult';
+
+type SingleReport = AggregatedResult & {
+  phabricatorReport: FormattedTestResult,
+  coverageMap: null,
+};
+
+type PhabricatorReport = {
+  phabricatorReport: Array<SingleReport>
+};
+
+const {formatTestResults} = require('jest-util');
+
+function summarize(coverageMap: CoverageMap) {
+  const summaries = Object.create(null);
+
+  coverageMap.files().forEach(file => {
+    const covered = [];
+    const lineCoverage = coverageMap.fileCoverageFor(file).getLineCoverage();
+
+    Object.keys(lineCoverage).forEach(lineNumber => {
+      // Line numbers start at one
+      const number = (parseInt(lineNumber, 10) - 1);
+      const visited = !!lineCoverage[lineNumber];
+      covered[number] = visited ? 'C' : 'U';
+    });
+
+    for (let i = 0; i < covered.length; i++) {
+      if (!covered[i]) {
+        covered[i] = 'N';
+      }
+    }
+
+    summaries[file] = covered.join('');
+  });
+
+  return summaries;
+}
+
+module.exports = function(results: AggregatedResult): PhabricatorReport {
+  let coverageMap;
+  if (results.coverageMap) {
+    coverageMap = summarize(results.coverageMap);
+  }
+
+  const formatter = (coverage, reporter) => coverageMap;
+  const report = formatTestResults(results, formatter);
+
+  return Object.assign(
+    (Object.create(null): any),
+    results,
+    {
+      coverageMap: null,
```

Comment:
Can you add a comment here indicating why we are setting this to null? (I'm assuming it is to preserve memory).